pkg/plugins/generic/generic_plugin.go

```diff
@@ -126,6 +124,31 @@ func (p *GenericPlugin) OnNodeStateChange(new *sriovnetworkv1.SriovNetworkNodeSt
 	return
 }
 
+// OnNodeStatusChange verify whether SriovNetworkNodeState CR status present changes on configured VFs.
+func (p *GenericPlugin) CheckStatusChanges(new *sriovnetworkv1.SriovNetworkNodeState) bool {
+	log.Log.Info("generic-plugin CheckStatusChanges()")
+
+	changed := false
+	for _, iface := range new.Spec.Interfaces {
+		found := false
+		for _, ifaceStatus := range new.Status.Interfaces {
+			if iface.PciAddress == ifaceStatus.PciAddress {
+				found = true
+				if utils.NeedUpdate(&iface, &ifaceStatus) {
```

Comment:
This does not cover all cases e,g kernel args. (all the more reason to not have yet another API to check changes)

Reply:
Indeed, it is missing that part. Thanks. I will add it.

However, I do believe we need another api. The logic of `needDrainNode` is different to what I have here. The check here starts from the interfaces of the Spec. It's simpler and faster. Same for the logic in `needRebootNode`, we don't to set the kernel args, but we need to check if the desiredKernelArgs parameter still matches. It' simpler.

Reply:
OnNodeStateChange should not "Apply" anything during the call.
current implementation has drifted from this, we should fix imo.